src/Swift.Runtime/src/Swift/SwiftArray.cs

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Swift;
+using Swift.Runtime;
+using Swift.Runtime.InteropServices;
+
+namespace Swift;
+
+/// <summary>
+/// Represents a Swift array payload.
+///
+/// The following diagram illustrates the hierarchy of a Swift array type.
+/// The actual implementation may differ:
+///
+///    struct Array
+///    +-----------------------------------------------------------------------+
+///    |   struct ArrayBuffer                                                 |
+///    |   +--------------------------------------------------------------+   |
+///    |   |   struct BridgeStorage                                       |   |
+///    |   |   +------------------------------------------------------+   |   |
+///    |   |   | var rawValue: IntPtr                                 |   |   |
+///    |   |   +------------------------------------------------------+   |   |
+///    |   +--------------------------------------------------------------+   |
+///    +-----------------------------------------------------------------------+
+/// </summary>
+[StructLayout(LayoutKind.Sequential, Size = 8)]
+public struct ArrayBuffer
+{
+    public IntPtr storage;
+}
+
+/// <summary>
+/// Represents a Swift array.
+/// </summary>
+/// <typeparam name="Element">The element type contained in the array.</typeparam>
+public class SwiftArray<Element> : IDisposable, ISwiftObject
+{
+    static nuint _payloadSize = SwiftObjectHelper<SwiftArray<Element>>.GetTypeMetadata().Size;
+    static nuint _elementSize = ElementTypeMetadata.Size;
+
+    // Swift array is a value type and doesn't contain an IntPtr payload
+    private ArrayBuffer buffer;
```

Comment:
```suggestion
    private ArrayBuffer _buffer;
```
Follow the coding style?